pkg/network/usm/sharedlibraries/watcher_test.go

```diff
@@ -186,26 +142,11 @@ func (s *SharedLibrarySuite) TestSharedLibraryDetectionWithPIDandRootNameSpace()
 	time.Sleep(10 * time.Millisecond)
 
 	// assert that soWatcher detected foo-libssl.so being opened and triggered the callback
-	require.Equal(t, libpath, pathDetected)
+	require.True(t, strings.HasSuffix(pathDetected, libpath))
```

Comment:
why did you switch from exact match to HasSuffix?

Reply:
wondering if we should use regex match here to ensure specific pattern
```go
fmt.Sprintf("/proc/(+\d)/root/%", libpath)
```

Reply:
I think it's fine to leave a simpler assertion here since I'm testing this more "in isolation" on the FileRegistry test suite:
https://github.com/DataDog/datadog-agent/pull/18355/files#diff-0d5db148a114a525cd8f679d11d07697be6bf6f1405733e97a9db66b2a39e1c8R108-R125